projects/kubernetes-sigs/etcdadm/patches/0014-Add-retry-on-listing-members-in-join-command.patch

```diff
@@ -0,0 +1,90 @@
+From 35a9d8d692c24b89be947ba216980676670668fa Mon Sep 17 00:00:00 2001
+From: Wonkun Kim <wonkun@amazon.com>
+Date: Fri, 3 Jun 2022 15:33:17 -0500
+Subject: [PATCH] Update join.go
+
+---
+ cmd/join.go | 51 +++++++++++++++++++++++++++++++--------------------
+ 1 file changed, 31 insertions(+), 20 deletions(-)
+
+diff --git a/cmd/join.go b/cmd/join.go
+index dee7eba9..fa503318 100644
+--- a/cmd/join.go
++++ b/cmd/join.go
+@@ -132,6 +132,22 @@ func membership() phase {
+ 				return nil
+ 			}
+ 
++			var lastErr error
++			retrySteps := 1
++			if in.etcdAdmConfig.Retry {
++				retrySteps = constants.DefaultBackOffSteps
++			}
++
++			// Exponential backoff for MemberList and MemberAdd (values exclude jitter):
++			// If --retry=false, add member only try one times, otherwise try five times.
++			// The backoff duration is 0, 2, 4, 8, 16 s
++			opts := wait.Backoff{
++				Duration: constants.DefaultBackOffDuration,
++				Steps:    retrySteps,
++				Factor:   constants.DefaultBackOffFactor,
++				Jitter:   0.1,
++			}
++
+ 			var localMember *etcdserverpb.Member
+ 			var members []*etcdserverpb.Member
+ 			log.Println("[membership] Checking if this member was added")
+@@ -139,13 +155,23 @@ func membership() phase {
+ 			if err != nil {
+ 				return fmt.Errorf("error checking membership: %v", err)
+ 			}
+-			ctx, cancel := context.WithTimeout(context.Background(), constants.DefaultEtcdRequestTimeout)
+-			mresp, err := client.MemberList(ctx)
+-			cancel()
++
++			err = wait.ExponentialBackoff(opts, func() (bool, error) {
++				ctx, cancel := context.WithTimeout(context.Background(), constants.DefaultEtcdRequestTimeout)
++				mresp, err := client.MemberList(ctx)
++				cancel()
++				if err != nil {
++					log.Warningf("[membership] Error listing members: %v, will retry after %s.", err, opts.Step())
```

Comment:
Is `[membership]` the right format for this warning message here?

Reply:
Yes, it indicates that the function name is `membership`.